sim-cli/src/parsing.rs

```diff
@@ -81,11 +86,33 @@ pub struct Cli {
     pub fix_seed: Option<u64>,
 }
 
+impl Cli {
+    pub fn validate(&self, sim_params: &SimParams) -> Result<(), anyhow::Error> {
+        // Validate that nodes and sim_graph are exclusively set
+        if !sim_params.nodes.is_empty() && !sim_params.sim_network.is_empty() {
+            return Err(anyhow!(
+                "Simulation file cannot contain {} nodes and {} sim_graph entries, simulation can only be run with real 
+            or simulated nodes not both.", sim_params.nodes.len(), sim_params.sim_network.len(),
+        ));
+        }
```

Comment:
nit: formatting looks odd here?

rustfmt doesn't work well inside of macros, maybe you can manually clean this up a bit?